src/Compilers/Test/Utilities/CSharp/CompilationTestUtils.cs

```diff
@@ -335,9 +335,9 @@ string toDisplayString(SyntaxNode syntaxOpt)
                     // We don't support VerifyTypes on suppressions at the moment
                     Assert.NotEqual(syntaxOpt.Kind(), SyntaxKind.SuppressNullableWarningExpression);
 
-                    return (syntaxOpt != null) && dictionary.TryGetValue(syntaxOpt, out var type)
-                        ? (type.IsNull ? "<null>" : type.ToDisplayString(TypeSymbolWithAnnotations.TestDisplayFormat))
-                        : null;
+                    return (syntaxOpt != null) && dictionary.TryGetValue(syntaxOpt, out var type) ?
+                        (!type.HasType ? "<null>" : type.ToDisplayString(TypeSymbolWithAnnotations.TestDisplayFormat)) :
```

Comment:
>(!type.HasType ? "<null>" : type.ToDisplayString(TypeSymbolWithAnnotations.TestDisplayFormat)) [](start = 24, length = 94)

do we need the conditional? (seems like `TSWA.ToDisplayString` can handle this case already)

Reply:
TSWA.ToDisplayString only handles it when there is no display format.  It checks `IsValueType` which will throw a null reference exception when there is a format.

---
In reply to: [260065526](https://github.com/dotnet/roslyn/pull/33648#discussion_r260065526) [](ancestors = 260065526)